### DISCUSSION TITLE
Batterie Infos (RS485 Schnittstelle)

### DISCUSSION THREAD

**Discussion by Manos1966** [skippermeister/OpenDTU-OnBattery#1] [Show and tell]
Batterie Daten (RS485 Schnittstelle)
Master:
![Master](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/c230c4d7-3862-41ef-a233-ba06f09564a6)

Slave-1:
![Slave](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/606483e4-c0c1-4e79-acc1-f598e93e1ae8)

**Comment by Manos1966** [comment]
![01](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/f7313b99-c468-4304-9416-2dfa186eb618)
![02](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/542250b2-0793-4a65-abb2-1a3449403dac)
![03](https://github.com/skippermeister/OpenDTU-OnBattery/assets/129660360/73c9f01d-d8ca-42da-a4da-9762ba6a8fc8)

**Comment by skippermeister** [comment]
Es gibt ein neues Release. Ich habe die Updates aus dem letzten Release von helgeerbe übernommen.

Das PowerMeter Modul hat jetzt eine eigenständige Task, die parallel zur Scheduler Loop läuft und das Hauptprogramm nicht mehr blockiert.

**Comment by DerHavey1994** [comment]
Ich würde meine Pylontech auch gerne zusätzlich mit diesem Fork & meinem alten ESP32 auslesen. Gibt es hierfür schon eine genaue Anleitung, wo man was anschließt? Von dem RS485 Modul kann ich mir ja eins aussuchen.

**Reply by skippermeister** [reply]
@DerHavey1994 
Dann nimm einfach dieses Pinmapping (daraus kannst Du die GPIOs ablesen):

Erstelle Dir daraus ein Pin_Mapping.json File und lade es in Dein ESP32, den Du zuvor mit meiner Software geflashed hast.

```
[
    {
        "name": "MyBoard NRF+Meanwell+PylonRS485",
        "nrf24": {
            "clk": 18,
            "cs": 5,
            "en": 2,
            "irq": 34,
            "miso": 19,
            "mosi": 23
        },
        "cmt": {},
        "display": {},
        "led": {},
        "victron": {
            "rs232_rx": 22,
            "rs232_tx": -1
        },
        "battery": {
            "rs485_rx": 16,
            "rs485_tx": 17,
            "rs485_rts": 4
        },
        "charger": {
            "can0_rx": 26,
            "can0_tx": 27
        },
        "batteryConnectedInverter": {
            "pre_charge": 32,
            "full_power": 33
        },
        "powermeter": {}
    }
]
```
falls Du keine MosFETS zwischen der Batterie und einem Hoymiles hast, dann kannst Du die Werte 
```
        "batteryConnectedInverter": {
            "pre_charge": -1,
            "full_power": -1
```
setzen.

Das Bin File 
[opendtu-onbattery-my_very_special_board.bin](https://github.com/skippermeister/OpenDTU-OnBattery/releases/download/2024.08.31/opendtu-onbattery-my_very_special_board.bin)
bzw.
[opendtu-onbattery-my_very_special_board.factory.bin](https://github.com/skippermeister/OpenDTU-OnBattery/releases/download/2024.08.31/opendtu-onbattery-my_very_special_board.factory.bin)
sollte direkt funktionieren.

**Reply by DerHavey1994** [reply]
@skippermeister 
![Screenshot 2024-09-05 144323](https://github.com/user-attachments/assets/eb41d478-7fdc-4521-b42e-8f413929dfbd)
So sieht es bei mir nun im Pin-Mapping aus, Batterie wird leider nicht angezeigt (habe auch schon gespeichert).
Verwendet habe ich den roten RS485 "Dongle". Muss ich den CAN oder RS485 Port vom Pylontech benutzen?

Firmware hatte ich diese drauf (Habe ja auch nur den ESP32 mit 4MB):
![Screenshot 2024-09-05 144429](https://github.com/user-attachments/assets/060547aa-38ee-4aa5-bf95-b2f60c5d5521)

**Reply by skippermeister** [reply]
Schick mir bitte das log vom Start des ESP32 bis ca 2min nach Start, wichtig ist die Bootphase. Und schicke mir das pinmapping. json File. Du musst einen Fehler im Pinmapping.json File haben.

**Reply by skippermeister** [reply]
Du solltest den unteren RS485 Port der Pylontech nutzen. Ich habe allerdings beide probiert und beide funktionieren bei mir.

**Reply by DerHavey1994** [reply]
[RS485 Mapping.json](https://github.com/user-attachments/files/16896538/RS485.Mapping.json)
[
    {
        "name": "Pylontech RS485",
        "nrf24": {
            "clk": 18,
            "cs": 5,
            "en": 2,
            "irq": 34,
            "miso": 19,
            "mosi": 23
        },
        "cmt": {},
        "display": {},
        "led": {},
        "victron": {},
        "battery": {
            "rs485_rx": 16,
            "rs485_tx": 17,
            "rs485_rts": 4
        },
        "charger": {},
        "batteryConnectedInverter": {
            "pre_charge": -1,
            "full_power": -1
        },
        "powermeter": {}
    }
]

**Reply by DerHavey1994** [reply]
![Screenshot 2024-09-05 172236](https://github.com/user-attachments/assets/2a85884c-1ec8-4313-839c-8eebc5c0ee2e)

hier bitte.

Als Info: Habe den NRF nur konfiguriert im Mapping. Ich benutze den ESP ausschließlich für die RS485 Abfrage.

**Reply by skippermeister** [reply]
Fehler gefunden im pinmapping parser. Sollte jetzt funktionieren.

**Reply by DerHavey1994** [reply]
Hat auf Anhieb funktioniert! Vielen Dank :)
Dachte es liegt an mir :-D

**Reply by skippermeister** [reply]
Super

**Reply by skippermeister** [reply]
@DerHavey1994 
Es gibt ein neues Release mit ein paar Korrekturen. Das hat zwar in Deinem Use-Case keine Auswirkung, aber falls Du doch ein Ladegerät über meine Software ansteuern möchtest, dann musst Du auf das neue release gehen.

Du findest Bin-Dateien für Meanwell und für Hauwei Ladegerät.
